File: src/test/java/voidchess/helper/ChessGameSupervisorMock.java
package voidchess.helper;

/**
 * @author stephan
 */
public class ChessGameSupervisorMock implements ChessGameSupervisor {
    private PawnPromotion defaultPawnTransform = PawnPromotion.QUEEN;

    public ChessGameSupervisorMock(PawnPromotion defaultPawnTransform) {
        this.defaultPawnTransform = defaultPawnTransform;
    }

    public PawnPromotion askForPawnChange(Position pawnPosition) {
        return defaultPawnTransform;
    }
}

File: src/test/kotlin/voidchess/board/ChessGameSupervisorMock.kt
package voidchess.board

import voidchess.board.ChessGameSupervisor
import voidchess.board.move.PawnPromotion
import voidchess.board.move.Position


class ChessGameSupervisorMock(private val defaultPawnTransform: PawnPromotion) : ChessGameSupervisor {

    override fun askForPawnChange(pawnPosition: Position): PawnPromotion {
        return defaultPawnTransform
    }
}
